services/harmony/app/models/label.ts

```diff
@@ -0,0 +1,70 @@
+import { Transaction } from '../util/db';
+
+/**
+ * Returns the labels for a given job
+ * @param transaction - the transaction to use for querying
+ * @param jobID - the UUID associated with the job
+ *
+ * @returns A promise that resolves to an array of strings, one for each label
+ */
+export async function getLabelsForJob(
+  transaction: Transaction,
+  jobID: string,
+): Promise<string[]> {
+  const query = transaction('jobs_labels')
+    .where({ job_id: jobID })
+    .orderBy(['jobs_labels.id'])
+    .innerJoin('labels', 'jobs_labels.label_id', '=', 'labels.id')
+    .select(['labels.value']);
+
+  const rows = await query;
+
+  return rows.map((row) => row.value);
+}
+
+/**
+ *  Set the labels for a given job/user. This is atomic - all the labels are set at once. Any
+ * existing labels are replaced.
+ * @param transaction - the transaction to use for querying
+ * @param jobID - the UUID associated with the job
+ * @param username - the username the labels belong to
+ * @param labels - the array of strings representing the labels. These will be forced to lower-case.
+ * If this is an empty array then any existing labels for the job will be cleared.
+ */
+export async function setLabelsForJob(
```

Comment:
I'm not positive how this is going to be used, but I do not like endpoints that require you to pass in what you think should exist plus any new info.

I'd want to have an endpoint to add one or more labels without changing any existing labels and an endpoint for removing one or more labels.

Reply:
The AC for this ticket:
```
* when a user submits a request they can submit one or more labels for the job
* the new label parameter is documented with examples
```
This ticket just allows the user to set labels when making a request. So not updating things. The updates will happen in later tickets. But one of the major use cases for the other tickets is to allow users to add labels from the workflow-ui, in which case I would think the the UI should show them the existing labels, which means you would have them and could resend the whole list. But if you want to do it piecemeal for some reason then that can be a different function.

Reply:
One thing I have tried to avoid is adding a full model for labels, since they are so simple.